website/project/decorators.py

```diff
@@ -378,3 +378,18 @@ def wrapped(*args, **kwargs):
 
     # Return decorated function
     return wrapped
+
+def http_error_if_disk_saving_mode(func):
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        _inject_nodes(kwargs)
```

Comment:
This statement is repeated in many decorators, which is unnecessary and inefficient. For example, in `website.project.views.drafts.register_draft_registration`, we call `inject_nodes` in three separate decorators iteratively.

Can we streamline this logic?